src/Npgsql.NodaTime/IntervalHandler.cs

```diff
@@ -60,6 +63,31 @@ public override void Write(Period value, NpgsqlWriteBuffer buf, NpgsqlParameter?
             buf.WriteInt32(value.Years * 12 + value.Months); // months
         }
 
+        Duration INpgsqlSimpleTypeHandler<Duration>.Read(NpgsqlReadBuffer buf, int len, FieldDescription? fieldDescription)
+        {
+            var microsecondsInDay = buf.ReadInt64();
+            var days = buf.ReadInt32();
+            var totalMonths = buf.ReadInt32();
+
+            if (totalMonths != 0)
+                throw new NpgsqlException("Cannot read PostgreSQL interval with non-zero months to NodaTime Duration. Try reading as a NodaTime Period instead.");
```

Comment:
From PostgreSQL docs:

> When this would result in a fractional number of months or days, the fraction is added to the lower-order fields using the conversion factors 1 month = 30 days and 1 day = 24 hours.

So why not to read months?

Reply:
While in *some* contexts PG will convert 1 month to 30 days, in others a month is a different number of days. For example:

```sql
SELECT '2020-02-01'::date + make_interval(months => 1);
```

... gives 2020-03-01, so the month was less than 30 days in that case.

The point of durations is to represent  an unambiguous, absolute amount of time, so it has no properties larger than a day (since those vary based on context). I think it would be quite dangerous and confusing to apply a 30 day conversion here.

Reply:
Yep, remember the problem with our types too.